recipes/caf/all/conanfile.py

```diff
@@ -25,61 +25,70 @@ class CAFConan(ConanFile):
     _build_subfolder = "build_subfolder"
     _cmake = None
 
-    @property
-    def _has_openssl(self):
-        return 'with_openssl' in self.options.values.keys() and self.options.with_openssl
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-            if self.settings.arch == "x86":
-                del self.options.with_openssl
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("actor-framework-" + self.version, self._source_subfolder)
 
     def requirements(self):
-        if self._has_openssl:
+        if self.options.with_openssl:
             self.requires("openssl/1.1.1j")
 
-    def configure(self):
-        if self.options.shared and self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
-        compiler_version = Version(self.settings.compiler.version.value)
-        if Version(self.version) > "0.17.6" and \
-            (self.settings.compiler == "gcc" and compiler_version < "7") or \
-                (self.settings.compiler == "clang" and compiler_version < "6") or \
-                (self.settings.compiler == "apple-clang" and compiler_version < "10") or \
-                (self.settings.compiler == "Visual Studio" and compiler_version < "16"):
-            raise ConanInvalidConfiguration("caf 0.18.0+ requires a C++17 compiler")
-        if self.settings.compiler == "gcc":
-            if compiler_version < "4.8":
-                raise ConanInvalidConfiguration("g++ >= 4.8 is required, yours is %s" % self.settings.compiler.version)
-        elif self.settings.compiler == "clang" and compiler_version < "4.0":
-            raise ConanInvalidConfiguration("clang >= 4.0 is required, yours is %s" % self.settings.compiler.version)
-        elif self.settings.compiler == "apple-clang" and compiler_version < "9.0":
-            raise ConanInvalidConfiguration("clang >= 9.0 is required, yours is %s" % self.settings.compiler.version)
-        elif self.settings.compiler == "apple-clang" and compiler_version > "10.0" and \
+    def _minimum_compilers_version(self, cppstd):
+        standards = {
+            "11": {
+                "Visual Studio": "15",
+                "gcc": "4.8",
+                "clang": "4",
+                "apple-clang": "9",
+            },
+            "17": {
+                "Visual Studio": "16",
+                "gcc": "7",
+                "clang": "6",   # Should be 5 but clang 5 has a bug that breaks compiling CAF
+                "apple-clang": "10",
+            },
+        }
+        return standards.get(cppstd) or {}
+
+    def validate(self):
+        cppstd = "11" if Version(self.version) <= "0.17.6" else "17"
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, cppstd)
+        min_version = self._minimum_compilers_version(cppstd).get(str(self.settings.compiler))
```

Comment:
The cpp std here collides with the conan concept, could it simply be based on the version?

Reply:
I was assuming that if they specify it in their profile then we should use that.  Otherwise use the recipes' standard versions.  But I'm happy to remove it too.

Reply:
But currently the `cppstd` you are using is hard-coded based on the recipe version 🤔 

The minimum compiler should be based on the version directly.

Reply:
The recipe version matches the version of the library.  Which version do you mean?

Reply:
`self.version`

Just my programmers brain trying to clean code. This suggestion is not required.


Right now its `version -> cpp std -> min compilers ` what I am suggesting is `version -> min compilers`

Reply:
Sounds good to me.